src/sentry/static/sentry/app/views/performance/table.tsx

```diff
@@ -55,56 +52,36 @@ type Props = {
   loadingProjects: boolean;
 };
 
-class Table extends React.Component<Props> {
-  renderResults(isLoading: boolean, tableData: TableData | null) {
-    let cells: React.ReactNode[] = [];
-    if (isLoading) {
-      return cells;
-    }
-    if (!tableData || !tableData.meta) {
-      return cells;
-    }
-    const columnOrder = this.props.eventView.getColumns();
-    const query = this.props.eventView.generateQueryStringObject();
-
-    tableData.data.forEach((row, index: number) => {
-      // check again to appease tsc
-      if (!tableData.meta) {
-        return;
-      }
-      cells = cells.concat(
-        this.renderRow(row, index, columnOrder, tableData.meta, query)
-      );
-    });
-    return cells;
-  }
+type State = {
+  widths: number[];
+};
 
-  renderRow(
-    row: TableDataRow,
-    rowIndex: number,
-    columnOrder: TableColumn<React.ReactText>[],
-    tableMeta: MetaType,
-    query: Query
-  ) {
-    const {organization, location, projects} = this.props;
+class Table extends React.Component<Props, State> {
+  state = {
+    widths: [],
+  };
 
-    return columnOrder.map((column, index) => {
-      const field = String(column.key);
-      // TODO(mark) add a better abstraction for this.
-      const fieldName = getAggregateAlias(field);
-      const fieldType = tableMeta[fieldName];
+  renderBodyCell = (tableMeta: TableData['meta']) => {
+    const {eventView, organization, projects, location} = this.props;
 
+    return (
+      column: TableColumn<keyof TableDataRow>,
+      dataRow: TableDataRow
+    ): React.ReactNode => {
+      if (!tableMeta) {
+        return null;
+      }
+      const field = String(column.key);
       const fieldRenderer = getFieldRenderer(field, tableMeta);
-      let rendered = fieldRenderer(row, {organization, location});
+      let rendered = fieldRenderer(dataRow, {organization, location});
 
-      const isFirstCell = index === 0;
+      if (field === 'transaction') {
+        const projectID = getProjectID(dataRow, projects);
 
-      if (isFirstCell) {
-        // the first column of the row should link to the transaction summary view
-        const projectID = getProjectID(row, projects);
+        const query = eventView.generateQueryStringObject();
         const target = transactionSummaryRouteWithQuery({
           orgSlug: organization.slug,
-          transaction: String(row.transaction) || '',
+          transaction: String(dataRow.transaction) || '',
```

Comment:
NB: Since it's existing code...

Is there a way to cast something as a string where it's not falsey and not empty string already? If `dataRow.transaction` is undefined/null, I think we just get those values as strings (which not sure if it's intended).

Reply:
I think we'd need to use ternaries or a utility function to do that.